spec/forms/developers_search_form_spec.rb

```diff
@@ -0,0 +1,89 @@
+require 'rails_helper'
+RSpec.describe DevelopersSearchForm do
+  describe '#search' do
+    before do
+      test_developer_one = create(:developer, email: 'tester@gmail.com')
+      test_developer_two = create(:developer, email: 'leluongtruong7793@gmail.com')
+      programming_language_one = create(:programming_language, name: 'PHP')
+      programming_language_two = create(:programming_language, name: 'Ruby')
+      create(:programming_language, name: 'Python')
+      language_one = create(:language, code: 'en')
+      language_two = create(:language, code: 'vn')
+      create(:developer_programming_language, developer_id: test_developer_one.id, programming_language_id: programming_language_one.id)
+      create(:developer_programming_language, developer_id: test_developer_two.id, programming_language_id: programming_language_two.id)
+      create(:developer_language, developer_id: test_developer_one.id, language_id: language_one.id)
+      create(:developer_language, developer_id: test_developer_one.id, language_id: language_two.id)
+      create(:developer_language, developer_id: test_developer_two.id, language_id: language_two.id)
+    end
+
+    context 'Test searching with no search fields provided' do
```

Comment:
@truongle7793 

Requirements
```
Implement a page that fulfills the following search conditions.
- Being able to search developers who can write Ruby.
- Being able to search developers who can write Ruby and speak English.
```
I think we should implement 2 test is enough.

Reply:
@thienhv 
Thanks you very much!
I have removed unnecessary system tests.